aws/sdk/examples/telephone-game/src/main.rs

```diff
@@ -0,0 +1,439 @@
+
+                    // Open the audio file with regular blocking IO File
+                    // rodio's Decoder requires stdlib Files
+                    let file = std::fs::File::open(&tmp_file_path).unwrap();
+                    let source =
+                        Decoder::new(std::io::BufReader::new(file)).expect("couldn't decode audio");
+
+                    // Set rodio to play the audio we just decoded
+                    sink.append(source);
+                    sink.sleep_until_end();
+
+                    info!("Did you hear it?");
+                }
+                Err(e) => error!("{}", e),
+            };
+
+            // Clear last iteration's line so we can receive new user input
+            line.clear();
+        }
+    })
+}
+
+
+/**
+# Play a game of Telephone w/ AWS
+
+1. Fetch user any user input that will override default values
+1. Create a config and required clients for AWS services
+1. Create a bucket to store audio and transcriptions if none exists
+1. Run in a loop
+    1. Start a speech synthesis task and set it to output to the previously created S3 bucket
+    1. Wait up to TASK_TIMEOUT_IN_SECONDS seconds for synthesis task to complete
+        - The status of the task is checked every TASK_WAIT_INTERVAL_IN_SECONDS in a loop
+        - Break out of the loop once the task succeeds or fails
+    1. Delete any pre-existing transcription job with the name "telephone-game-transcription"
+        - Job names must be unique so we clear the old job to reuse the name.
+    1. Wait up to TASK_TIMEOUT_IN_SECONDS seconds for transcripion job to complete
+        - The status of the job is checked every TASK_WAIT_INTERVAL_IN_SECONDS in a loop
+        - Break out of the loop once the job succeeds or fails
+    1. Download the transcript JSON file from S3 and print what Transcribe heard
+    1. Break out of the loop if an error occurs or we reach the iteration limit
+1. Log a final message before closing
+    - If everything succeeded, log the original phrase, the current phrase, and how many iterations occurred
+    - If a failure occurs, log the error and any errors that caused it
+*/
+fn play_telephone(matches: &ArgMatches) {
+    let rt = Runtime::new().expect("failed to create an async runtime");
+    let number_of_iterations = matches
+        .value_of("iterations")
+        .and_then(|i| i.parse::<u32>().ok())
+        .unwrap_or(DEFAULT_NUMBER_OF_ITERATIONS);
+
+    if number_of_iterations == 0 {
+        error!("Iterations must be a number greater than 0");
+        return;
+    }
+
+    let original_phrase = matches.value_of("phrase").unwrap_or_default();
+    let mut current_phrase = original_phrase.to_owned();
+
+    let bucket_name = matches
+        .value_of("s3_bucket_name")
+        .map(ToOwned::to_owned)
+        .unwrap_or_else(|| "telephone-game".to_owned());
+
+    let result = rt.block_on(async {
+        let config = aws_config::load_from_env().await;
```

Comment:
probably a little easier to follow if this is split off into a separate async fn that is invoked with `rt.block_on`